Runner/Runner.py

```diff
@@ -0,0 +1,141 @@
+        base_url=_BATCH_ACCOUNT_URL,
+        subscription_id=_BATCH_ACCOUNT_SUB)
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "TestConfig",
+        help="A manifest file that contains a list of all the jobs and pools you want to create.")
+    args = parser.parse_args()
+
+    # "Tests/TestConfiguration.json"
+    jsonConfigFile = args.TestConfig
+
+    # Clean up any storage container that is older than a 7 days old.
+    try:
+        Utils.cleanup_old_resources(blob_client)
+    except Exception as e:
+        raise e
+
+    try:
+        with open(jsonConfigFile) as f:
+            template = json.load(f)
+
+        for i in range(0, len(template["tests"])):
+            jobSettings = template["tests"][i]
+
+            applicationLicenses = None
+            try:
+                applicationLicenses = jobSettings["applicationLicense"]
+            except KeyError as e:
+                pass
+
+            _job_managers.append(
+                JobManager.JobManager(
+                    jobSettings["template"],
+                    jobSettings["poolTemplate"],
+                    jobSettings["parameters"],
+                    jobSettings["expectedOutput"],
+                    applicationLicenses))
+
+        images_refernces = []
+        for i in range(0, len(template["images"])):
+            image = template["images"][i]
+            images_refernces.append(
+                Utils.ImageReference(
+                    image["osType"],
+                    image["offer"],
+                    image["version"]))
+
+        Utils.logger.info("{} jobs will be created".format(len(_job_managers)))
+        loop = asyncio.get_event_loop()
+        loop.run_until_complete(asyncio.gather(*[j.upload_assets(blob_client) for j in _job_managers]))
+        Utils.logger.info("Creating pools...")
+        loop.run_until_complete(asyncio.gather(*[j.create_pool(batch_client, images_refernces) for j in _job_managers]))
+        Utils.logger.info("Submitting jobs...")
+        loop.run_until_complete(asyncio.gather(*[j.create_and_submit_Job(batch_client) for j in _job_managers]))
+        Utils.logger.info("waiting for jobs to complete...")
+        loop.run_until_complete(asyncio.gather(*[j.wait_for_tasks_to_complete(batch_client, _timeout) for j in _job_managers]))
+
+    except batchmodels.batch_error.BatchErrorException as err:
+        traceback.print_exc()
+        Utils.print_batch_exception(err)
+        raise
+    finally:
+        loop = asyncio.get_event_loop()
+        # Delete all the jobs and containers needed for the job
+        # Reties any jobs that failed
+        Utils.logger.info("-----------------------------------------")
+        loop.run_until_complete(asyncio.gather(*[j.retry(batch_client, blob_client, _timeout/2) for j in _job_managers]))
+        #loop.run_until_complete(asyncio.gather(*[j.delete_resouces(batch_client, blob_client) for j in _job_managers]))
```

Comment:
💡 Please note these are commented lines, if no longer in need, then please delete them.

Reply:
They need to be uncommented in the final PR, This is done to speed up testing.